### ISSUE TITLE
Enumeration inconsistency in smfc configuration

### ISSUE THREAD

**Issue by petersulyok** [petersulyok/smfc#3]
The issue: `smfc` configuration defines the list of the hard disks and the adequate list of the temperature files in `sys/hwmon` system in the following way:
```
hd_names=/dev/sda /dev/sdb /dev/sdc /dev/sdd /dev/sde /dev/sdf /dev/sdg /dev/sdh
hwmon_path=/sys/class/scsi_device/0:0:0:0/device/hwmon/hwmon*/temp1_input
           /sys/class/scsi_device/1:0:0:0/device/hwmon/hwmon*/temp1_input
           /sys/class/scsi_device/2:0:0:0/device/hwmon/hwmon*/temp1_input
           /sys/class/scsi_device/3:0:0:0/device/hwmon/hwmon*/temp1_input
           /sys/class/scsi_device/4:0:0:0/device/hwmon/hwmon*/temp1_input
           /sys/class/scsi_device/5:0:0:0/device/hwmon/hwmon*/temp1_input
           /sys/class/scsi_device/6:0:0:0/device/hwmon/hwmon*/temp1_input
           /sys/class/scsi_device/7:0:0:0/device/hwmon/hwmon*/temp1_input
```
Unfortunately both lists are created and enumerated in the Linux Kernel independently at boot time, so the identical order and proper pairing are not guaranteed (and could be different after a reboot). This problem will have different consequences in different hardware configurations. Here are two examples:

1. There are 8 HDs used in a RAID array and the OS is booted from a NVME SSD. In this case smfc is working properly (i..e. the reading temperature of HDs, and _Standby Guard_ feature is working properly) despite of inconsistency of the lists above.
2.  There are 5 HDs installed, four of them are organized in a RAID array, OS is booted from the fifth HD. In this case the inconsistency of the lists will generate real problems. Reading of the temperatures of the HDs and the _Standby Guard_ feature can mix HDs inside and outside of the RAID array.

**Comment by petersulyok** [comment]
Solution proposal:

1. Configuration item `hd_names=` should be defined with stable device names like `/dev/disk/by-id/`. These name are not changing after reboots.
2. Configuration item `hwmon_path=` will be generated automatically in a consistent way. `smfc` will find the a proper temperature file in `sys/hwmon` system for a specific device on the `hd_names=` list. In this way the items will be paired properly on both list. The configuration item can be removed from the configuration file.

This change might generate slightly increased startup time for the service, but no further performance impacts are expected.

**Comment by petersulyok** [comment]
Design:

In the following way we can find the adequate `hwmon` files for a specified disk in the configuration:
- The `hd_names=` parameter must be one or more `/dev/disk/by-id/...` reference(s)
- These are links to SATA device names (e.g. `../../sda`), for further identification we have this save this value
- Reading of the files of directory `/sys/class/scsi_disk/` we can find the previously found SATA name. It can be found in directory `/sys/class/scsi_disk/?:0:0:0/device/block/`
- If we found the proper SATA name then the `hwmon_path` will be `/sys/class/scsi_disk/?:0:0:0/device/hwmon/hwmon*/temp1_input`.

In this way `hwmon_path=` configuration parameter will be optional, will be generated automatically based on the value of `hd_names=` parameter.

**Comment by petersulyok** [comment]
This issue is fixed in [commit 9b8c40eb76b6ff448f9f96e95b77484707926731](https://github.com/petersulyok/smfc/commit/9b8c40eb76b6ff448f9f96e95b77484707926731).
